Objects/regions.c

```diff
@@ -997,110 +1315,31 @@ static PyObject *add_to_region(PyObject *obj, Py_region_ptr_t region)
     Py_RETURN_NONE;
 }
 
+static void PyRegion_dealloc(PyRegionObject *self) {
+    // Name is immutable and not in our region.
 
-static bool is_bridge_object(PyObject *op) {
-    Py_region_ptr_t region = Py_REGION(op);
-    if (IS_LOCAL_REGION(region) || IS_IMMUTABLE_REGION(region)) {
-        return false;
+    // The object region has already been reset.
+    // We now need to update the RC of our metadata field.
+    if (self->metadata) {
+        self->metadata->bridge = NULL;
+        regionmetadata_dec_rc(self->metadata);
+        self->metadata = NULL;
```

Comment:
I wonder if this should be:
```suggestion
        regionmetadata* tmp = self->metadata;
        self->metadata = NULL;
        regionmetadata_dec_rc(tmp);
```
Not sure there is an issue, but this pattern is generally safer.